src/device-manager.ts

```diff
@@ -0,0 +1,160 @@
+export interface AudioDevice {
+  deviceId: string;
+  label: string;
+  default: boolean;
+  current: boolean;
+}
+
+export interface DeviceManagerCallbacks {
+  onDeviceError: (error: Error) => void;
+  onDeviceDisconnected?: (deviceId: string) => void;
+  onDeviceSwitched?: (fromDeviceId: string, toDeviceId: string) => void;
+}
+
+export interface DeviceManagerOptions {
+  /** Enable automatic device switching when current device is disconnected */
+  autoSwitchOnDisconnect?: boolean;
+  /** Enable device change event listening */
+  listenForDeviceChanges?: boolean;
+}
+
+interface MediaDeviceWithDefault extends MediaDeviceInfo {
+  default?: boolean;
+}
+
+export class DeviceManager {
+  private wavRecorder: any; // WavRecorder instance
+  private callbacks: DeviceManagerCallbacks;
+  private currentDeviceId: string | null = null;
+  private options: Required<DeviceManagerOptions>;
+  private deviceChangeListener: ((devices: MediaDeviceWithDefault[]) => void) | null = null;
+
+  constructor(wavRecorder: any, callbacks: DeviceManagerCallbacks, options: DeviceManagerOptions = {}) {
+    this.wavRecorder = wavRecorder;
+    this.callbacks = callbacks;
+    this.options = {
+      autoSwitchOnDisconnect: options.autoSwitchOnDisconnect ?? true,
+      listenForDeviceChanges: options.listenForDeviceChanges ?? true,
+    };
+
+    this.updateCurrentDeviceId();
+
+    if (this.options.listenForDeviceChanges) {
+      this._setupDeviceChangeListener();
+    }
+  }
+
+  /**
+   * Updates the current device ID from the active audio stream
+   */
+  updateCurrentDeviceId(): void {
+    try {
+      const currentStream = this.wavRecorder.getStream();
+      const currentTrack = currentStream?.getAudioTracks()[0];
+      if (currentTrack) {
+        this.currentDeviceId = currentTrack.getSettings().deviceId || null;
+      }
+    } catch (error) {
+      // Silent fail - device ID will be null
+    }
+  }
+
+  /**
+   * Handles errors consistently
+   */
+  private handleError(message: string, error: unknown): Error {
+    const err = error instanceof Error ? error : new Error(String(error));
+    console.error(message, err);
+    this.callbacks.onDeviceError(err);
+    return err;
+  }
+
+  /**
+   * Sets up the device change event listener
+   */
+  private _setupDeviceChangeListener(): void {
+    this.deviceChangeListener = async (devices: MediaDeviceWithDefault[]) => {
+      try {
+        // Ensure we have current device ID
+        if (!this.currentDeviceId) {
```

Comment:
Why do we need to call updateCurrentDeviceId here, when it's already called in the constructor.